src/components/ui/input.tsx

```diff
@@ -0,0 +1,25 @@
+import * as React from "react"
+
+import { cn } from "@/lib/utils"
+
+export interface InputProps
+  extends React.InputHTMLAttributes<HTMLInputElement> {}
+
+const Input = React.forwardRef<HTMLInputElement, InputProps>(
+  ({ className, type, ...props }, ref) => {
+    return (
+      <input
+        type={type}
+        className={cn(
```

Comment:
are these coming from shadcn? They should already be present in the `shad` folder

Reply:
yes these are coming from shadcn

Reply:
i thought the shad files were in the ui folder in components